specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-06-20-preview/HybridCompute.json

```diff
@@ -2266,6 +2474,123 @@
       ],
       "description": "The List hybrid machine operation response."
     },
+    "LicenseProfile": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/LicenseProfileProperties",
+          "description": "Hybrid Compute License Profile properties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Describes a license profile in a hybrid machine."
+    },
+    "LicenseProfileUpdate": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/LicenseProfileUpdateProperties",
+          "description": "Describes License Profile Update Properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceUpdate"
+        }
+      ],
+      "description": "Describes a Machine Extension Update."
+    },
+    "EsuKey": {
+      "type": "object",
+      "properties": {
+        "sku": {
+          "type": "string",
+          "description": "SKU number."
+        },
+        "licenseStatus": {
+          "type": "string",
+          "enum":[
+            "1",
+            "0"
+          ],
+          "description": "The current status of the license profile key."
+        }
+      }
+    },
+    "EsuStatus": {
+      "type": "string",
+      "enum":[
+        "enrolled",
+        "unenrolled",
+        "ineligible"
+      ],
+      "description": "The ESU enrollment state."
+    },
+    "EsuStatusUpdateRecord": {
+      "type": "object",
+      "properties": {
+        "oldState": {
+          "$ref": "#/definitions/EsuStatus",
+          "description": "The old ESU enrollment state."
+        },
+        "newState": {
+          "$ref": "#/definitions/EsuStatus",
+          "description": "The new ESU enrollment state."
+        },
+        "date": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time that the state has been changed."
+        }
+      }
+    },
+    "LicenseProfileProperties": {
```

Comment:
NIT: Would it be better to keep the top-level namespace cleaner to allow for future expansion?

instead of:
LicenseProfileProperties
{
arcEsuEnrolled : "",
esuKeys: "",
esuHistory: ""
}

do:
LicenseProfileProperties
{
esu : {
  arcEsuEnrolled : "",
  esuKeys: "", 
  esuHistory: ""
  }
}